lib/ruby_lsp/requests/formatting.rb

```diff
@@ -56,6 +58,8 @@ def formatted_file
         else
           SyntaxTree.format(@document.source)
         end
+      rescue RuboCop::Runner::InfiniteCorrectionLoop => error
```

Comment:
I am afraid this is still not the correct place. In this file, you are still not guaranteed that `RuboCop` will exist as a constant.

The error handling needs to live inside the new `RuboCopRunner`, since that's the only location that does a require and bails if the require fails.